### PR TITLE
chore: rename `denomination_trace` event type to `denomination`

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

## Summary by CodeRabbit

- **Documentation**
  - Updated documentation to reflect the renaming of `denomination_trace` to `denomination`.

- **Refactor**
  - Renamed `EmitDenomTraceEvent` function to `EmitDenomEvent`.
  - Updated event attributes to use the new denomination naming convention.
  - Changed event type from `EventTypeDenomTrace` to `EventTypeDenom` in relevant functions.

### CODE REVIEW

File: modules/apps/transfer/keeper/relay.go [hunk 1]
```diff
@@ -275,7 +275,7 @@ func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data t
 
 		ctx.EventManager().EmitEvent(
 			sdk.NewEvent(
-				types.EventTypeDenomTrace,
+				types.EventTypeDenom,
 				sdk.NewAttribute(types.AttributeKeyTraceHash, token.Denom.Hash().String()),
 				sdk.NewAttribute(types.AttributeKeyDenom, voucherDenom),
```
Comment (colin-axner) [inline modules/apps/transfer/keeper/relay.go:280]: probably should update this to be `token.Denom` instead of `voucherDenom`, otherwise it's redundant

**Review by colin-axner — APPROVED** [review]

**Review by damiannolan — APPROVED** [review]

File: docs/docs/02-apps/01-transfer/05-events.md [hunk 2]
```diff
@@ -28,7 +28,7 @@ slug: /apps/transfer/events
 | fungible_token_packet | amount        | \{amount\}      | 
 | fungible_token_packet | success       | \{ackSuccess\}  | 
 | fungible_token_packet | memo          | \{memo\}        | 
-| denomination_trace    | trace_hash    | \{hex_hash\}    | 
+| denomination          | trace_hash    | \{hex_hash\}    |
```
Comment (crodriguezvega) [inline docs/docs/02-apps/01-transfer/05-events.md:31]: I am going to update this before merging.

File: modules/apps/transfer/internal/events/events.go [hunk 3]
```diff
@@ -103,13 +103,13 @@ func EmitOnTimeoutEvent(ctx sdk.Context, packetData types.FungibleTokenPacketDat
...
+// EmitDenomEvent emits a denomination event in the OnRecv callback.
+func EmitDenomEvent(ctx sdk.Context, token types.Token) {
 	ctx.EventManager().EmitEvent(
 		sdk.NewEvent(
-			types.EventTypeDenomTrace,
-			sdk.NewAttribute(types.AttributeKeyTraceHash, traceHash),
-			sdk.NewAttribute(types.AttributeKeyDenom, voucherDenom),
+			types.EventTypeDenom,
+			sdk.NewAttribute(types.AttributeKeyDenomHash, token.Denom.Hash().String()),
+			sdk.NewAttribute(types.AttributeKeyDenom, token.Denom.String()),
```
Comment (crodriguezvega) [inline modules/apps/transfer/internal/events/events.go:112]: @DimitrisJim Should we JSON-encode as well this value?

Comment (colin-axner) [reply modules/apps/transfer/internal/events/events.go:112]: yes definitely

**Pull request merged**